internal/v1/handler.go

```diff
@@ -86,17 +88,25 @@ func (h *Handlers) GetDistributions(ctx echo.Context) error {
 	dr := h.server.distroRegistry(ctx)
 	userID, err := h.server.getIdentity(ctx)
 	if err != nil {
-		return err
+		return echo.NewHTTPError(http.StatusInternalServerError, fmt.Sprintf("Failed to get user identity: %v", err))
 	}
 
+	distributions, err := h.filterDistributions(userID.OrgID(), *dr)
+	if err != nil {
+		return echo.NewHTTPError(http.StatusInternalServerError, "Failed to filter distributions").SetInternal(err)
+	}
+	return ctx.JSON(http.StatusOK, distributions)
+}
+
+func (h *Handlers) filterDistributions(orgID string, distroMap distribution.DistroRegistry) (DistributionsResponse, error) {
```

Comment:
Nit: I'd probably keep using a pointer, just because we've got a pointer in the caller.
But given the DistroRegistry structure is just a single pointer, it's competely ok to copy it as well, so it's just a question of taste, nothing really important to address.